docker/README.md

```diff
@@ -1,39 +1,92 @@
-# Trino Gateway Docker Image
+# Trino Gateway Docker image
 
-## About the Container
+The Docker image of Trino Gateway is designed for the following use cases:
 
-This Docker image is designed to:
+* Manual usage in front of Trino clusters by mounting in a configuration file
+* Automated usage with an orchestration system like Kubernetes to simplify
+  deployment
 
-* be spun up in front of Trino clusters by mounting in a configuration file
-* simplify deployment into an orchestration system
-
-## Quickstart
-
-### Dependencies
+## Build requirements
 
 This docker build process requires:
 
 * [Docker Compose V2](https://docs.docker.com/compose/)
 * jq
 
-### Run the Trino Gateway server
+# Building a custom Docker image
+
+To build an image for a locally modified version of Trino Gateway, run the Maven
+build and then build the image for your desired processor architecture.
+
+```bash
+./build.sh arm64
+```
+
+By default, the scripts builds all valid processor architectures `amd64`,
+`arm64`, and `ppc64le`:
+
+```bash
+./build.sh
+```
+
+The Docker build process prints the ID of the image, which is also tagged with 
+`trino-gateway:xxx-SNAPSHOT-yyy`, where `xxx-SNAPSHOT` is the version number and
+`-yyy` is the processor architecture.
+
+To build an image for a specific released version of Trino Gateway, specify the 
+`-r` option. The build script downloads all the required artifacts:
+
+```bash
+./build.sh -r 4
+```
+
+Use the `-h` option for further help.
 
-You can launch the Trino Gateway and relevant dependencies through docker for testing purposes.
+### Run Trino Gateway
+
+You can launch Trino Gateway and required PostgreSQL for testing purposes with
+the following command examples.
+
+Use a locally-built image on a ARM-based machine, such as a Macbook laptop.
+
+```bash
+export TEST_GATEWAY_IMAGE="trino-gateway:6-SNAPSHOT-arm64"
+export TEST_PLATFORM="arm64"
+```
+
+Use a locally-built image on a AMD64-based machine, such as a typical Windows
+desktop or laptop.
+
+```bash
+export TEST_GATEWAY_IMAGE="trino-gateway:6-SNAPSHOT-amd64"
+export TEST_PLATFORM="amd64"
+```
+
+Use a published image for AMD64 from DockerHub.
 
 ```bash
-# Replace these variables to match your test requirements
-# Ex: You may be locally building the docker image so something like
-# `trino-gateway:5-SNAPSHOT-amd64` might be what you're expecting
+export TEST_GATEWAY_IMAGE="trinodb/trino-gateway:latest"
+export TEST_PLATFORM="amd64"
+```
+
+The release process publishes images to DockerHub for Trino Gateway 6 and newer.
+
+Set the default image
+```bash
+export TRINO_GATEWAY_IMAGE=${TEST_GATEWAY_IMAGE} 
+export DOCKER_DEFAULT_PLATFORM=${TEST_PLATFORM}
+
 TEST_GATEWAY_IMAGE="trinodb/trino-gateway:latest"
 TEST_PLATFORM="amd64"
```

Comment:
nitpick: duplicative, can be deleted

Reply:
This are all alternative configs so each one has to be complete..